[WIP] First proposal of the 'Jenkins & Java' page

batmat opened the pull request:
Couldn't test it locally, gradle failing currently.

WDYT?

rtyler commented on content/solutions/java.html.haml:
```diff
@@ -37,3 +37,39 @@ plugins:
 ---
+
+:markdown
+Jenkins supports building Java projects since its inception, and for a reason!
```
The content under a `:markdown` declaration should be indented two spaces. HAML is whitespace sensitive

rtyler commented on content/solutions/java.html.haml:
```diff
@@ -37,3 +37,39 @@ plugins:
+%center
+  %img{:src => '/images/solution-images/jenkins-maven-step.png'}/
+
+## Gradle
```
The `:markdown` declaration must be repeated before markdown starts

rtyler commented on content/solutions/java.html.haml:
```diff
@@ -37,3 +37,39 @@ plugins:
+%center
+%img{:src => '/images/solution-images/jenkins-gradle-step.png'}/
+
+## Using Ant
```
As far as I'm aware, using Ant is definitely not "best practice" anymore. Android was the last big project using Ant that i was aware of, and they have since switched to Gradle.

I think we should stick to talking about Gradle and Maven with regards to Java projects.

batmat replied:
Absolutely. Even worse is using a custom script 😊. I would go berserk if I
still had to work on such project in 2016.

But I had actually put my opinion and so on aside as I just didn't think
this page was meant to be a best practice reference.

I'll remove the ant and custom script parts.

Thanks for the feedback
Le 29 janv. 2016 8:10 AM, "R. Tyler Croy" <notifications@github.com> a
écrit :

> In content/solutions/java.html.haml
> <https://github.com/jenkins-infra/jenkins.io/pull/62#discussion_r51231612>
> :
>
> > +## Apache Maven
> > +In any FreeStyle job, as *currently* Maven is supported in standard, you can use the
> > +dedicated step. One advantage is, as for all Jenkins tools, that you can select a specific
> > +Maven version and have Jenkins automatically install it on the build node it's gonna run on.
> > +
> > +%center
> > +  %img{:src => '/images/solution-images/jenkins-maven-step.png'}/
> > +
> > +## Gradle
> > +
> > +As the associated plugin is not installed per defaut, first install the
> > +
> > +%center
> > +%img{:src => '/images/solution-images/jenkins-gradle-step.png'}/
> > +
> > +## Using Ant
>
> As far as I'm aware, using Ant is definitely not "best practice" anymore.
> Android was the last big project using Ant that i was aware of, and they
> have since switched to Gradle.
>
> I think we should stick to talking about Gradle and Maven with regards to
> Java projects.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkins-infra/jenkins.io/pull/62/files#r51231612>.
>

rtyler commented:
@batmat The Gradle errors you saw were likely Haml compilation errors, since the template is actually invalid Haml

The images, would you mind ensuring that screenshots are no wider than 600px? This will make them fit nicely, without resizing, in the legacy site.

batmat commented:
Yup. Will look into it. Sorry for the garbage dump. Wanted to gather
feedback on the content.
Le 29 janv. 2016 8:11 AM, "R. Tyler Croy" <notifications@github.com> a
écrit :

> @batmat <https://github.com/batmat> The Gradle errors you saw were likely
> Haml compilation errors, since the template is actually invalid Haml
>
> The images, would you mind ensuring that screenshots are no wider than
> 600px? This will make them fit nicely, without resizing, in the legacy site.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkins-infra/jenkins.io/pull/62#issuecomment-176608902>
> .
>

rtyler commented:
@batmat I don't think the solution pages should be "best practices" pages for Programming in general, but using Gradle/Maven in the context of Jenkins I believe is definitely a best practice and will result in a richer experience for the Jenkins user. That's why I suggested it, IMHO we should suggest best practices for integrating in _Jenkins_.

batmat commented:
@rtyler force pushed with:
* fixed haml so that `gradlew assemble` works locally
* fixed images so that there are 600 px wide max (note that some existing ones had already more than 600 px, but I didn't touch them
* removed the ant & custom script part.

HTH

rtyler commented:
@batmat thanks! I'm going to go ahead and merge this so we have something on the site for today. We can continue to iterate on it once more folks start to see it and "[ Improve this page ]" :)
